test/e2e/runner/infra.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+
+	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
+)
+
+// TestnetInfraAPI provides an API for interacting with the infrastructure for
+// an entire testnet.
+type TestnetInfraAPI interface {
+	// GenerateConfig generates any necessary configuration for the
+	// infrastructure provider during testnet setup.
+	GenerateConfig() error
```

Comment:
```suggestion
	Setup() error
```

`GenerateConfig` is pretty specific and the underlying functionality may need to grow beyond that. I would push this a bit further though, is there anything that is done in this method that cannot safely happen in the constructor or is this something that we may need to do multiple times? If not, I think we should fold this into the constructor and document that callers should ensure that `Cleanup` is called when they are done using the type.